src/mlpack/core/math/random.hpp

```diff
@@ -78,6 +78,44 @@ inline int RandInt(const int lo, const int hiExclusive)
 }
 
 /**
+ * Generates a random integer from range [0, N) with given probabilities
+ * P(0), P(1), ..., P(N-1) (arbitrary discrete distribution).
+ * 
+ * @param weights Probabilities of individual numbers. Note that the function
+ *                accepts unnormalized probabilties as long as they are
+ *                non-negative and sum to a positive number.
+ * @return A random integer sampled from the specified distribution.
+ */
+inline int RandInt(const arma::vec& weights)
+{
+  // Build cumulative probabilities from event probabilities.
+  std::vector<double> slots(weights.size());
+  double runningSum = 0;
+  for (size_t i = 0; i < weights.size(); ++i) {
```

Comment:
I was thinking if that check is necessary, since it was made clear in the documentation, we could simplify the expression:

```
arma::vec slots = cumsum(weights);
slots /= arma::accu(weights);
return std::lower_bound(slots.begin(), slots.end(), Random()) - slots.begin();
```

Let me know what you think.